spring-cloud-kubernetes-jaeger/src/main/java/org/springframework/cloud/kubernetes/jaeger/JaegerKubernetesAutoConfiguration.java

```diff
@@ -0,0 +1,97 @@
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.utils.Utils;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.autoconfig.TracerAutoConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.kubernetes.discovery.KubernetesServiceInstance;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.Assert;
+
+@Configuration
+@AutoConfigureBefore(TracerAutoConfiguration.class)
+@EnableConfigurationProperties(KubernetesJaegerDiscoveryProperties.class)
+public class JaegerKubernetesAutoConfiguration {
+
+	private static final Log LOG = LogFactory.getLog(JaegerKubernetesAutoConfiguration.class);
+
+    @Bean
+    public Tracer tracer(KubernetesClient client, KubernetesJaegerDiscoveryProperties discoveryProperties) throws Exception {
+
+    	String serviceName = discoveryProperties.getServiceName();
+    	String traceServerName = discoveryProperties.getTracerServerName();
+		String serviceNamespace = Utils.isNotNullOrEmpty(discoveryProperties.getServiceNamespace()) ? discoveryProperties.getServiceNamespace() : client.getNamespace();
+
+        List<ServiceInstance> services = getInstances(client, traceServerName, serviceNamespace);
+        String serviceUrl = services.stream()
+                .findFirst()
+                .map(s -> s.getUri().toString())
+                .orElse(null);
+
+        if (serviceUrl == null || serviceUrl.isEmpty()) {
+			LOG.info("Jaeger k8s starter creating Noop Tracer");
+            return NoopTracerFactory.create();
+        }
+
+        return  jaegerTracer(serviceUrl, serviceName);
+    }
+
+    private static List<ServiceInstance> getInstances(KubernetesClient client, String name, String namespace) {
+        Assert.notNull(name, "[Assertion failed] - the service name must not be null");
+
+        return Optional.ofNullable(client.endpoints().inNamespace(namespace).withName(name).get())
+                .orElse(new Endpoints())
+                .getSubsets()
+                .stream()
+                .flatMap(s -> s.getAddresses().stream().map(a -> (ServiceInstance) new KubernetesServiceInstance(name, a ,s.getPorts().stream().findFirst().orElseThrow(IllegalStateException::new), false)))
+                .collect(Collectors.toList());
+    }
+
+	private Tracer jaegerTracer(String serviceURL, String serviceName) throws Exception {
+		URL url = new URL(serviceURL);
+		Sender sender = new UdpSender(url.getHost(), url.getPort(), 0);
+		return new com.uber.jaeger.Tracer.Builder(serviceName,
+			new RemoteReporter(sender, 100, 50,
```

Comment:
Some of these properties should be configurable, see https://github.com/uber/jaeger-client-java/tree/master/jaeger-core#configuration-via-environment at least queue size and flush interval.